sdk/download.go

```diff
@@ -50,15 +52,38 @@ func TarballURL(version string) string {
 	return u.String()
 }
 
-func DownloadFile(file, url string) error {
-	plog.Infof("Downloading %s to %s", url, file)
+func DownloadFile(file, fileURL string, client *http.Client) error {
+	plog.Infof("Downloading %s to %s", fileURL, file)
 
+	// handle bucket urls by using api to get media link
+	parseURL, err := url.Parse(fileURL)
+	if err != nil {
+		return err
+	}
+	if parseURL.Scheme == "gs" {
+		if client == nil {
+			return fmt.Errorf("DownloadFile requires non-nil client with gs:// links")
+		}
+		api, err := storage.New(client)
+		if err != nil {
+			plog.Fatal(err)
+		}
+		path := strings.TrimLeft(parseURL.Path, "/")
+		obj, err := api.Objects.Get(parseURL.Host, path).Do()
+		if err != nil {
+			plog.Fatal(err)
+		}
+		plog.Noticef("%v", fileURL)
+		fileURL = obj.MediaLink
```

Comment:
Interesting, iirc last time we tried using media link directly and passing it to the go http client go screwed up escaping and broke the link. I wonder if go or the GS service has been fixed.